projects/plugins/jetpack/extensions/blocks/videopress/tracks-editor.js

```diff
@@ -0,0 +1,433 @@
+		const trackToDelete = tracks[ trackIndex ];
+		setIsDeletingTrackIndex( trackIndex );
+		deleteTrackForGuid( trackToDelete, guid ).finally( () => {
+			onChange( tracks.filter( ( _track, index ) => index !== trackIndex ) );
+			setIsDeletingTrackIndex( -1 );
+		} );
+	};
+
+	let content;
+	if ( tracks.length === 0 ) {
+		content = (
+			<p className="videopress-block-tracks-editor__tracks-informative-message">
+				{ __(
+					'Tracks can be subtitles, captions, chapters, or descriptions. They help make your content more accessible to a wider range of users.',
+					'jetpack'
+				) }
+			</p>
+		);
+	} else {
+		content = tracks.map( ( track, index ) => {
+			return (
+				<div key={ index } className="videopress-block-tracks-editor__track-list-track">
+					<span>{ track.label }</span>
+					<div className="videopress-block-tracks-editor__track-list-track-delete">
+						<Button
+							isLink
+							isDestructive
+							onClick={ () => {
+								onDeleteTrack( index );
+							} }
+							aria-label={ sprintf(
+								/* translators: %s: Label of the video text track e.g: "French subtitles" */
+								__( 'Delete %s', 'jetpack' ),
+								track.label
+							) }
+							disabled={ isDeletingTrackIndex === index }
+						>
+							{ isDeletingTrackIndex === index
+								? __( 'Deleting', 'jetpack' )
+								: __( 'Delete', 'jetpack' ) }
+						</Button>
+					</div>
+				</div>
+			);
+		} );
+	}
+
+	return (
+		<MenuGroup
+			label={ __( 'Text tracks', 'jetpack' ) }
+			className="videopress-block-tracks-editor__track-list"
+		>
+			{ content }
+		</MenuGroup>
+	);
+}
+
+function SingleTrackEditor( { track, guid, onChange, onClose, onCancel, trackExists } ) {
+	const [ errorMessage, setErrorMessage ] = useState();
+	const [ isSavingTrack = false, setIsSavingTrack ] = useState();
+	const { label = '', srcLang = '', kind = DEFAULT_KIND } = track;
+
+	const fileName = track.tmpFile ? track.tmpFile.name : '';
+
+	const mediaUpload = useSelect( select => {
+		return select( blockEditorStore ).getSettings().mediaUpload;
+	}, [] );
+
+	if ( ! mediaUpload ) {
+		return null;
+	}
+
+	return (
+		<NavigableMenu>
+			<div className="videopress-block-tracks-editor__single-track-editor">
+				<span className="videopress-block-tracks-editor__single-track-editor-label">
+					{ __( 'Edit track', 'jetpack' ) }
+				</span>
+				<div className="videopress-block-tracks-editor__single-track-editor-edit-file-label">
+					<span>{ __( 'File', 'jetpack' ) }:</span>
+					{ '' !== fileName && (
+						<span className="videopress-block-tracks-editor__single-track-editor-edit-file-label-name">
+							<strong>{ fileName }</strong>
+						</span>
+					) }
+					<MediaUploadCheck>
+						<FormFileUpload
+							onChange={ event => {
+								const files = event.target.files;
+
+								if ( ! files.length > 0 ) {
+									return;
+								}
+
+								track.tmpFile = files[ 0 ];
+								onChange( track );
+							} }
+							accept=".vtt,text/vtt"
```

Comment:
I wonder if we should surface the accepted formats a bit more, so folks know what to upload? Maybe as a title at least, something like:

```diff
diff --git a/projects/plugins/jetpack/extensions/blocks/videopress/tracks-editor.js b/projects/plugins/jetpack/extensions/blocks/videopress/tracks-editor.js
index 81e28f4640..bbf2f9fa4b 100644
--- a/projects/plugins/jetpack/extensions/blocks/videopress/tracks-editor.js
+++ b/projects/plugins/jetpack/extensions/blocks/videopress/tracks-editor.js
@@ -24,7 +24,7 @@ import { useSelect } from '@wordpress/data';
 import { useState } from '@wordpress/element';
 
 const DEFAULT_KIND = 'subtitles';
-
+const ACCEPTED_FILE_EXTENSIONS = 'vtt,text/vtt';
 const KIND_OPTIONS = [
 	{ label: __( 'Subtitles', 'jetpack' ), value: 'subtitles' },
 	{ label: __( 'Captions', 'jetpack' ), value: 'captions' },
@@ -224,7 +224,11 @@ function SingleTrackEditor( { track, guid, onChange, onClose, onCancel, trackExi
 								track.tmpFile = files[ 0 ];
 								onChange( track );
 							} }
-							accept=".vtt,text/vtt"
+							accept={ ACCEPTED_FILE_EXTENSIONS }
+							title={ sprintf(
+								__( 'Accepted file formats: %s', 'jetpack' ),
+								ACCEPTED_FILE_EXTENSIONS
+							) }
 							render={ ( { openFileDialog } ) => {
 								return (
 									<Button

```

Reply:
I like it! Unfortunately the `title` attr gets eaten, but what about a help label like this:
<img width="358" alt="Screen Shot 2021-11-03 at 3 04 39 PM" src="https://user-images.githubusercontent.com/789137/140192613-3022a50b-79a2-4230-ba47-a015ef869933.png">

Reply:
Yes, that looks nice to me (I'm no designer though :) )!